packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/reducer.js

```diff
@@ -103,8 +103,16 @@ const reducer = (state, action) =>
         if (replace) {
           set(draftState, [...path, 'connect'], [value]);
         } else {
-          const nextValue = get(draftState, [...path, 'connect']);
-          nextValue.push(value);
+          const savedRelations = get(state, [...path, 'results']);
+          const existInSavedRelation =
+            savedRelations &&
+            savedRelations?.findIndex((savedRelations) => savedRelations.id === value.id) !== -1;
```

Comment:
Since you are using optional chaining here, you can simplify this to 
```suggestion
          const existInSavedRelation =
            savedRelations?.findIndex((savedRelations) => savedRelations.id === value.id) !== -1 ?? false;
```